src/realm/object-store/schema.cpp

```diff
@@ -349,20 +366,34 @@ std::vector<SchemaChange> Schema::compare(Schema const& target_schema, SchemaMod
     return changes;
 }
 
-void Schema::copy_keys_from(realm::Schema const& other) noexcept
+void Schema::copy_keys_from(realm::Schema const& other, bool is_schema_additive) noexcept
 {
+    // compute properties for objects that are in common between the current schema and the new schema.
+    // Append to the end of the new schema peristed properties, all those properties that have been deleted
     zip_matching(*this, other, [&](ObjectSchema* existing, const ObjectSchema* other) {
         if (!existing || !other)
             return;
 
+        std::vector<Property> unmatched_properties;
         existing->table_key = other->table_key;
         for (auto& current_prop : other->persisted_properties) {
             auto target_prop = existing->property_for_name(current_prop.name);
             if (target_prop) {
                 target_prop->column_key = current_prop.column_key;
             }
+            else if (is_schema_additive) {
+                unmatched_properties.push_back(current_prop);
+            }
+        }
+
+        if (is_schema_additive) {
+            existing->persisted_properties.insert(existing->persisted_properties.end(), unmatched_properties.begin(),
```

Comment:
Just out of curiosity, would it be better for the SDKs to have a `dummy` property/schema_object to signal that all the properties/objects behind that point do not actually match with the user specified schema?